contracts/compressor/IntentCompressor.sol

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IInbox} from "../interfaces/IInbox.sol";
+import {IIntentSource} from "../interfaces/IIntentSource.sol";
+import {Intent, Call, TokenAmount, Route, Reward} from "../types/Intent.sol";
+
+struct EncodedIntent {
+    uint8 sourceChainIndex;
+    uint8 destinationChainIndex;
+    // Reward token
+    uint8 rewardTokenIndex;
+    uint48 rewardAmount;
+    // Route token
+    uint8 routeTokenIndex;
+    uint48 routeAmount;
+    // Expiry duration
+    uint24 expiryDuration;
+    // Salt
+    bytes8 salt;
+}
+
+struct EncodedFulfillment {
+    uint8 sourceChainIndex;
+    uint8 destinationChainIndex;
+    // Reward token
+    uint8 routeTokenIndex;
+    uint48 routeAmount;
```

Comment:
Is the assumption here that we round all token amounts to 6 decimals? 
USDC on BNB has 18 decimal 😭 
We should have the mapping for decimals for each token.

Reply:
You can bump this type as you still have room in bytes32
`uint64 routeAmount;`

Reply:
Before merging, let's make sure we are using all the bytes. 👌🏻 

We may need another solution because, for 18 decimals, using 64 bytes will give us a maximum of 18 units (e.x. 18.44... USDC). We could use a float approach with some precision plus an exponent. 

This approach wouldn't work for very precise amount but could work for USD tokens with 18 decimals.

Reply:
This is solved here https://github.com/eco/eco-routes/pull/209#discussion_r1985958273
By having encoding decimals. Could be 9 decimals for most practical cases.